File: Flutter/Vistian.Remoting.Package/lib/src/parameter.dart
/*
  A parameter passed to a remoting service.
 */
import 'dart:convert';
import 'package:json_annotation/json_annotation.dart';

part 'parameter.g.dart';

@JsonSerializable(createFactory: false)
class Parameter{
  // the name of the parameter
  @JsonKey(name:'Name')
  final String name;
  // the actual value to be used for the parameter
  @JsonKey(name:'Value')
  final String value;
  // the value of the parameter
  @JsonKey(name:'ValueType')
  final String valueType;

  Parameter(this.name,dynamic value,this.valueType):value = json.encode(value);

  Map<String,dynamic> toJson() => _$ParameterToJson(this) ;
}

File: Xamarin/Vistian.Flutter.Remoting.Core/Parameter.cs
using System;
namespace Vistian.Flutter.Remoting
{
    public class Parameter
    {
        /// <summary>
        /// Get or set the parameter name
        /// </summary>
        /// <value>The name.</value>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the value.
        /// </summary>
        /// <value>The value.</value>
        public object Value { get; set; }

        /// <summary>
        /// Get or set the Dart Value Type
        /// </summary>
        /// <value>The type of the value.</value>
        public string ValueType { get; set; }
    }
}
